code/game/gamemodes/nuclear/nuclear.dm

```diff
@@ -35,14 +35,17 @@ proc/issyndicate(mob/living/M as mob)
 	if(possible_syndicates.len < 1)
 		return 0
 
-	if(LAZYLEN(possible_syndicates) > agents_possible)
-		agent_number = agents_possible
+
+
+	if(num_players() >= required_players && num_players() < required_players + 20)
+		agent_number = 3
+	else if(num_players() >= required_players + 20 && num_players() < required_players + 40)
+		agent_number = 4
 	else
-		agent_number = possible_syndicates.len
+		agent_number = 5
 
-	var/n_players = num_players()
-	if(agent_number > n_players)
-		agent_number = n_players/2
+	if(LAZYLEN(possible_syndicates) < agent_number)
+		agent_number = LAZYLEN(possible_syndicates)
```

Comment:
`possible_syndicates` is not a lazy list. Can just use `length(possible_syndicates)`